aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -617,57 +617,31 @@ TORCH_IMPL_FUNC(bmm_out_cuda)(const Tensor& batch1, const Tensor& batch2, const
   }
 }
 
-namespace {
-
-inline void dot_check(const Tensor& self, const Tensor& other) {
-  TORCH_CHECK(
-      self.dim() == 1 && other.dim() == 1,
-      "1D tensors expected, but got ",
-      self.dim(),
-      "D and ",
-      other.dim(),
-      "D tensors");
-  TORCH_CHECK(
-      self.scalar_type() == other.scalar_type(),
-      "dot : expected both vectors to have same dtype, but found ",
-      self.scalar_type(),
-      " and ",
-      other.scalar_type());
-  TORCH_CHECK(
-      self.numel() == other.numel(),
-      "inconsistent tensor size, expected tensor [",
-      self.numel(),
-      "] and src [",
-      other.numel(),
-      "] to have the same number of elements, but got ",
-      self.numel(),
-      " and ",
-      other.numel(),
-      " elements respectively");
-  TORCH_CHECK(
-      (self.numel() <= INT_MAX) && (self.stride(0) <= INT_MAX) &&
-          (other.stride(0) <= INT_MAX),
-      "dot only supports n, incx, incy with the bound [val] <= %d",
-      INT_MAX);
-}
+// Early definition as dot and vdot call each other.
+void vdot_out_cuda_impl(const Tensor& self, const Tensor& other, const Tensor& result);
 
-} // anonymous namespace
 
-Tensor dot_cuda(const Tensor& self, const Tensor& other) {
+void dot_out_cuda_impl(const Tensor& self, const Tensor& other, const Tensor& result) {
   if (self.is_complex()) {
     if (self.is_conj()) {
       if (other.is_conj()) {
-        return (dot_cuda(self.conj(), other.conj())).conj();
-       } else {
-         return vdot_cuda(self.conj(), other);
-       }
+        dot_out_cuda_impl(self.conj(), other.conj(), result);
```

Comment:
Is the `.conj()` on the result lost here?

Reply:
I think I've resolved this